### PR TITLE
Docs for atomic orderings: link to the 'nomicon article for further reading

### PR DESCRIPTION
**Pull request by DirkyJerky**

### CODE REVIEW

File: src/libcore/sync/atomic.rs [hunk 1]
```diff
@@ -140,10 +140,15 @@ unsafe impl<T> Sync for AtomicPtr<T> {}
 /// to be moved either before or after the atomic operation; on the other end
 /// "relaxed" atomics allow all reorderings.
 ///
+/// If you are confused or don't have enough time to research which ordering to use, use `SeqCst`.
+/// Of all the options it has the most unsurpising effect (see the nomicon for details. [1]) 
+/// The downside is you miss out on several optimizations the other orderings offer.
+///
```
Comment (DirkyJerky) [inline src/libcore/sync/atomic.rs]: This is just something completely optional I feel would help people (beginners) who have no clue how orderings work and just want to get sample code working.

**Comment by DirkyJerky** [comment]
Tidy failed because:
Rustfmt failed at iter/mod.rs:1231: line exceeded maximum length (maximum: 100, found: 102) (sorry)
Rustfmt failed at str/mod.rs:1643: line exceeded maximum length (maximum: 100, found: 145) (sorry)
Rustfmt failed at str/mod.rs:1668: line exceeded maximum length (maximum: 100, found: 143) (sorry)
Rustfmt failed at fmt/mod.rs:1103: line exceeded maximum length (maximum: 100, found: 116) (sorry)

**Comment by DirkyJerky** [comment]
Also apoligies if this has been discussed before, but would a `Default` implementation (of SeqCst) be appropriate for `Ordering`? (as how cpp does it)

File: src/libcore/sync/atomic.rs (hunk 1 above)
Comment (nagisa) [inline src/libcore/sync/atomic.rs]: I’d argue that if one is confused/doesn’t have time to research, they should not be writing code involving atomics at all and just use a `Mutex`.

**Comment by BurntSushi** [comment]
@steveklabnik I assigned this over to you because you might have opinions on the wording here.

**Comment by BurntSushi** [comment]
@DirkyJerky A `Default` impl doesn't feel completely right or completely wrong to me. I expect that should be submitted as a separate PR, and that would need to go through the libs team approval.

File: src/libcore/sync/atomic.rs (hunk 1 above)
Comment (DirkyJerky) [inline src/libcore/sync/atomic.rs]: Very good point

**Comment by steveklabnik** [comment]
looks good to me, thanks!

@bors: r+ rollup

**Pull request merged**